From 01334589f1eae801b4ed8fb72fe3816ad5b0fe78 Mon Sep 17 00:00:00 2001
From: Nick Mathewson <nickm@torproject.org>
Date: Thu, 10 Dec 2015 20:02:22 -0500
Subject: [PATCH 4/5] Simplify cpd_opts usage.

---
 src/or/config.c | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/src/or/config.c b/src/or/config.c
index d9ba774..c6d4de3 100644
--- a/src/or/config.c
+++ b/src/or/config.c
@@ -1187,12 +1187,11 @@ options_act_reversible(const or_options_t *old_options, char **msg)
   }
 
   /* Ensure data directory is private; create if possible. */
-  cpd_check_t cpd_group_opts = CPD_NONE;
+  cpd_check_t cpd_opts = running_tor ? CPD_CREATE : CPD_CHECK;
   if (options->DataDirectoryGroupReadable)
-      cpd_group_opts = CPD_GROUP_READ;
+      cpd_opts |= CPD_GROUP_READ;
   if (check_private_dir(options->DataDirectory,
-                        running_tor ?
-                        CPD_CREATE|cpd_group_opts : CPD_CHECK|cpd_group_opts,
+                        cpd_opts,
                         options->User)<0) {
     tor_asprintf(msg,
               "Couldn't access/create private data directory \"%s\"",
-- 
2.5.0

