From 4b0e9fff27acec544615810138e678869fcdd280 Mon Sep 17 00:00:00 2001
From: Nick Mathewson <nickm@torproject.org>
Date: Thu, 10 Dec 2015 20:03:37 -0500
Subject: [PATCH 5/5] Fix wide line; log why chmod failed.

---
 src/or/config.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/or/config.c b/src/or/config.c
index c6d4de3..7b42c9f 100644
--- a/src/or/config.c
+++ b/src/or/config.c
@@ -1205,7 +1205,8 @@ options_act_reversible(const or_options_t *old_options, char **msg)
   if (options->DataDirectoryGroupReadable) {
     /* Only new dirs created get new opts, also enforce group read. */
     if (chmod(options->DataDirectory, 0750)) {
-      log_warn(LD_FS,"Unable to make %s group-readable.", options->DataDirectory);
+      log_warn(LD_FS,"Unable to make %s group-readable: %s",
+               options->DataDirectory, strerror(errno));
     }
   }
 #endif
-- 
2.5.0

